docs/developer/architecture/ci.rst

```diff
@@ -0,0 +1,120 @@
+CI
+==
+
+Pre merge
+---------
+
+What should be tested in pre-merge on every ``user/*``, ``feature/*``,
+``improvement/*``, ``bugfix/*``, ``w/*`` branches.
+
+- Building tests:
+
+  - Build
+  - Lint
+  - Unit tests
+
+- Installation tests
+
+  - Simple install RHEL
+  - Simple install CentOs + expansion
+
+When merging several PRs at the same time, so when we are on a ``q/*``
+branches, we may want to have additionnal test like:
+
+- Simple upgrade/downgrade
+
+Pre-merge shouldn't be too long as it run on every push on the feature
+branches.
+
+.. _post-merge:
+
+Post merge
+----------
+
+On each and every ``development/2.*`` branches we want to run complex tests to
+ensure that the current developmnent product is well working.
```

Comment:
Change to `development` or simply have `On each and every ``development/2.*`` branches we want to run complex tests to ensure that the current product continues to work`??